packages/react/src/time-picker/TimePicker.tsx

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react';
+import { useCombobox } from 'downshift';
+import { type ComboboxProps } from '../combobox/Combobox';
+import { ComboboxBase } from '../combobox/ComboboxBase/ComboboxBase';
+import {
+	type DefaultComboboxOption,
+	useComboboxInputId,
+} from '../combobox/utils';
+import { type TimeFormat } from '../time-input/utils';
+import { filterOptions, generateOptions } from './utils';
+
+export function TimePicker({
+	id,
+	inputRef: inputRefProp,
+	interval = 15,
+	loading = false,
+	max = '24:00',
+	maxWidth = 'md',
+	min = '00:00',
+	onChange,
+	timeFormat = 'h:mm aaa',
+	value,
+	...props
+}: TimePickerProps) {
+	const {
+		// We remove this prop so that it can't be passed to ComboboxBase
+		// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+		// @ts-ignore
+		renderItem,
+		...consumerProps
+	} = props;
```

Comment:
Instead of excluding `renderItem` here, what do you think about adding `renderItem={undefined}` after the spread of props on line 85? You could also add some TS help to the prop types:

```ts
// In the component
{...props}
renderItem={undefined}

// In the prop types

/** Item rendering is handled internally by TimePicker. */
renderItem?: never;
```

You do run the risk of people thinking you like TS 😉

Reply:
I have `renderItem` being `Omit`ted, is that sufficient? That feels nicer to me?

Not sure I'm a fan of spreading props which could have `renderItem` and then manually overriding it. Not sure what the React internals are, but it feels like double-work in my head

Reply:
>I have renderItem being Omitted, is that sufficient? That feels nicer to me?

That's the most important part, for sure. The `never` with JSDoc would be explicitly calling out and explaining why within the prop types.

>Not sure I'm a fan of spreading props which could have renderItem and then manually overriding it. Not sure what the React internals are, but it feels like double-work in my head

I don't think it would be any more double work than destructuring would it? And it co-locates the props object modification with its purpose.

Another thing that would be a benefit is removing the `@ts-ignore`. Having said that, if you wanted to keep this destructuring approach, we should replace the `@ts-ignore` with `@ts-expect-error We remove this prop so that it can't be passed to ComboboxBase` and remove the two lines above it.